Implemented RSA encryptor

pragyanmehrotra opened the pull request:
Hi, @rafaeelaudibert I took your advice and tried to code according to as the other encryptors were coded, the catch is the input text for decrypt should be in hex.

This has reference to Issue #19

rafaeelaudibert commented on encryptor/encryptors/rsa.py:
```diff
@@ -0,0 +1,85 @@
+
+class Rsa:
+
+	Default_E = 65537
```
Please, name these different default parameters in uppercase, i.e.: `DEFAULT_E`

rafaeelaudibert commented on encryptor/encryptors/rsa.py:
```diff
@@ -0,0 +1,85 @@
+
+	@staticmethod
+	def encrypt(text: str, n: int = Default_N, e: int = Default_E):
+		if (n is None):
```
Remove the parenthesis around the condition

rafaeelaudibert commented on encryptor/encryptors/rsa.py:
```diff
@@ -0,0 +1,85 @@
+			except ValueError:
+				raise WrongTypeParameter("n", int, type(n))
+
+		if (e is None):
```
Remove the parenthesis around the condition

rafaeelaudibert commented on encryptor/encryptors/rsa.py:
```diff
@@ -0,0 +1,85 @@
+	Default_N = Default_P*Default_Q
+
+	@staticmethod
+	def encrypt(text: str, n: int = Default_N, e: int = Default_E):
```
You could use Python ternary operator in the if checks below, like the following snippet:

```python3
try:
  n = DEFAULT_N if n is None else int(n)
except ValueError:
  raise WrongTypeParameter("n", int, type(n))
```

rafaeelaudibert commented on encryptor/encryptors/rsa.py:
```diff
@@ -0,0 +1,85 @@
+			except ValueError:
+				raise WrongTypeParameter("e", int, type(e))
+
+		message = int(hexlify(bytes(text,'utf-8')),16)
```
Please add spaces between the function parameters, for all the called functions below

rafaeelaudibert commented on encryptor/encryptors/rsa.py:
```diff
@@ -0,0 +1,85 @@
+		return hex(encrypted_message)[2:].split('L')[0]
+
+	@staticmethod
+	def decrypt(text: str, p: int = Default_P, q: int = Default_Q, e: int = Default_E):
```
Same thing with ternary operator as explained above

rafaeelaudibert commented on encryptor/encryptors/rsa.py:
```diff
@@ -0,0 +1,85 @@
+	def modinv(a, m):
+		g, x, y = Rsa.egcd(a, m)
+		if g != 1:
+			raise Exception("Modular Inverse does not exist")
```
You may consider creating a custom exception for this

pragyanmehrotra commented:
Hi, @rafaeelaudibert I have incorporated the changes you have asked for and added a few test cases as well, Please have a look at them and lemme know if any further changes are needed.

rafaeelaudibert commented:
Hey @pragyanmehrotra , just passing by to see if you have any updates on that?

pragyanmehrotra commented:
Hi, @rafaeelaudibert Sorry for the late reply and a bit of inactivity I had a lot of quizzes this week and was a little busy with deadlines I'll drop the changes by this weekend.

rafaeelaudibert commented:
Hey @pragyanmehrotra, don't worry! Be free to take the time you need to take for it! Good luck with the deadlines.

pragyanmehrotra commented:
Yeah got a heavy weekend, had to implement the whole GUI part and the animations of a Game as a part of the deadline due Sunday night, Fortunately was able to get something done in time. Have a look at the tests I have added for the api and lemme know if some further changes are needed

rafaeelaudibert commented:
That's awesome! Thank you very much @pragyanmehrotra, have a good end of the year!
